### ISSUE TITLE
Create TrainableAnnotator subclass of Annotator

### ISSUE THREAD

**Issue by cmalinmayor** [funkelab/funtracks#20]
- [ ] Takes the project as an argument, parameters, the same as Annotator
    - [ ] Can look at candidate graph to get the pinned nodes/edges and use the project to get any additional information needed to train the model (e.g. crops of the raw data)
- [ ] Training function
    - [ ] After each iteration:
         - [ ] Subclasses/implementations need to implement a function that given a training data set (as a pinned candidate graph), trains one "iteration", and then returns the loss.
        - [ ] Send progress signal with the loss
        - [ ] check for stopping training
        - [ ] check for updating training data
 - [ ] Save and load model from/to file implemented in subclass
 - [ ] Same as Annotator, needs `compute` and `update` functions for predicting for all nodes/edges and predicting for one node/edge
